lib/rexml/parsers/baseparser.rb

```diff
@@ -211,8 +211,9 @@ def pull_event
         #STDERR.puts @source.encoding
         #STDERR.puts "BUFFER = #{@source.buffer.inspect}"
         if @document_status == nil
+          pos = @source.pos
```

Comment:
How about renaming `pos` to `start_position` or something to indicate this is the position before we parsing the next event?